explorer-backend/transactions/watcher.go

```diff
@@ -27,6 +27,30 @@ func NewWatcher(logger *zap.Logger, configs *common.BridgeConfig, db *mongo.Data
 	}
 }
 
+func (w *Watcher) GetLatestEventIndexAlephium(ctx context.Context) (*uint32, error) {
+	return w.GetLatestEventIndex(ctx, vaa.ChainIDAlephium)
+}
+
+func (w *Watcher) GetLatestEventIndexEth(ctx context.Context) (*uint32, error) {
+	return w.GetLatestEventIndexEvm(ctx, vaa.ChainIDEthereum, 64)
+}
+
+func (w *Watcher) GetLatestEventIndexBsc(ctx context.Context) (*uint32, error) {
+	return w.GetLatestEventIndexEvm(ctx, vaa.ChainIDBSC, 15)
+}
+
+func (w *Watcher) GetLatestEventIndexEvm(ctx context.Context, chainId vaa.ChainID, confirmations uint32) (*uint32, error) {
+	eventIndex, err := w.GetLatestEventIndex(ctx, chainId)
+	if err != nil {
+		return nil, err
+	}
+	fromIndex := uint32(0)
+	if *eventIndex > confirmations {
+		fromIndex = *eventIndex - confirmations
```

Comment:
Do we need `fromIndex = max(1, *eventIndex - confirmations` so it's always positive?

Reply:
The `fromIndex >=0` is always true, and since we use `uint32` here, if `eventIndex` is less than `confirmations`, then it will underflow, and we will get a big `uint32` number.

If we want `max(1, *eventIndex - confirmations)`, we need to do type conversion: `fromIndex := uint32(maxInt64(1, int64(*eventIndex) - int64(confirmations)))`.

Reply:
And although golang already supports generics, there is still no elegant way to write a generic comparison function.

Reply:
Then we should probably do something like:
```
if (*eventIndex <= confirmations) 1 else (*eventIndex - confirmations)
```